homeassistant/components/media_player/universal.py

```diff
@@ -0,0 +1,433 @@
+        _LOGGER.warning(
+            'Universal Media Player (%s) children not list in config. '
+            'They will be ignored.',
+            config[CONF_NAME])
+        config[CONF_CHILDREN] = []
+
+
+def validate_commands(config):
+    """ validate commands """
+    if CONF_COMMANDS not in config:
+        config[CONF_COMMANDS] = {}
+    elif not isinstance(config[CONF_COMMANDS], dict):
+        _LOGGER.warning(
+            'Universal Media Player (%s) specified commands not dict in '
+            'config. They will be ignored.',
+            config[CONF_NAME])
+        config[CONF_COMMANDS] = {}
+
+
+def validate_attributes(config):
+    """ validate attributes """
+    if CONF_ATTRS not in config:
+        config[CONF_ATTRS] = {}
+    elif not isinstance(config[CONF_ATTRS], dict):
+        _LOGGER.warning(
+            'Universal Media Player (%s) specified attributes '
+            'not dict in config. They will be ignored.',
+            config[CONF_NAME])
+        config[CONF_ATTRS] = {}
+
+    for key, val in config[CONF_ATTRS].items():
+        config[CONF_ATTRS][key] = val.split('|', 1)
+
+
+class UniversalMediaPlayer(MediaPlayerDevice):
+    """ Represents a universal media player in HA """
+    # pylint: disable=too-many-public-methods
+
+    def __init__(self, hass, name, children, commands, attributes):
+        # pylint: disable=too-many-arguments
+        self.hass = hass
+        self._name = name
+        self._children = children
+        self._cmds = commands
+        self._attrs = attributes
+
+        track_state_change(hass, self.dependencies, self.update_state)
+
+    def _entity_lkp(self, entity_id=None, state_attr=None, state_as_obj=True):
+        """ Looks up an entity state from hass """
+        if entity_id is None:
+            return
+
+        state_obj = self.hass.states.get(entity_id)
+
+        if state_obj is None:
+            return
+
+        if state_attr:
+            return state_obj.attributes.get(state_attr)
+        if state_as_obj:
+            return state_obj
+        return state_obj.state
+
+    def _override_or_child_attr(self, attr_name):
+        """ returns either the override or the active child for attr_name """
+        if attr_name in self._attrs:
+            return self._entity_lkp(*self._attrs[attr_name],
+                                    state_as_obj=False)
+
+        return self._child_attr(attr_name)
+
+    def _child_attr(self, attr_name):
+        """ returns the active child's attr """
+        active_child = self.active_child_state
+        return active_child.attributes.get(attr_name) if active_child else None
+
+    def _override_or_child_service(self, service_name, **service_data):
+        """ calls either a specified or active child's service """
+        if service_name in self._cmds:
+            cmd_data = self._cmds[service_name]
+            domain, service = cmd_data[CONF_SERVICE].split('.', 1)
+            self.hass.services.call(domain, service,
```

Comment:
Please use the service helper [`call_from_config`](https://github.com/balloob/home-assistant/blob/dev/homeassistant/helpers/service.py#L14). It's the same code that powers calling services from config for the automation and alexa components.